### PR TITLE
Refactor/eventbridge_v2: Remove duplicated tests in rule file

### PR DESCRIPTION
**Pull request by maxhoheiser**
## Motivation
Testing the rule API is already fully covered in the API tests in `test_events`, therefore the tests in `test_events_rules` are redundant and are therefore deleted.

### CODE REVIEW

File: tests/aws/services/events/test_events_rules.py [hunk 1]
```diff
@@ -1,345 +0,0 @@
...
-    clean_up(
-        bus_name=event_bus_name,
-        rule_name=rule_name,
-        target_ids=target_id,
-        queue_url=queue_url,
-    )
-
-
-@markers.aws.validated
-@pytest.mark.skip
```
Comment (joe4dev) [inline tests/aws/services/events/test_events_rules.py:308]: I guess this test was flaky and therefore skipped 🧹 

It seems we are still missing tests for the CloudWatch target in `test_events_targets.py`, right?

File: tests/aws/services/events/test_events_rules.py [hunk 2]
```diff
@@ -1,345 +0,0 @@
...
-    snapshot.match("list-rules", response)
-    rules = response["Rules"]
-    assert len(rules) == 1
-    assert json.loads(rules[0]["EventPattern"]) == TEST_EVENT_PATTERN
-
-    clean_up(rule_name=rule_name)
-
-
-@markers.aws.validated
-def test_rule_disable(aws_client, clean_up):
```
Comment (joe4dev) [inline tests/aws/services/events/test_events_rules.py:42]: better coverage in `aws.services.events.test_events.TestEventRule.test_disable_re_enable_rule` 💯 
🧹🧹🧹

File: tests/aws/services/events/test_events_rules.py [hunk 3]
```diff
@@ -1,345 +0,0 @@
...
-from tests.aws.services.events.helper_functions import (
-    assert_valid_event,
-    is_v2_provider,
-    sqs_collect_messages,
-)
-from tests.aws.services.events.test_events import TEST_EVENT_PATTERN
-
-
-@markers.aws.validated
-def test_put_rule(aws_client, snapshot, clean_up):
```
Comment (joe4dev) [inline tests/aws/services/events/test_events_rules.py:23]: Much better coverage in `aws.services.events.test_events.TestEventRule` 🥳 
🧹

**Review by joe4dev — COMMENTED** [review]
Good initiative to clean up redundant tests now that we're having better coverage 💯 🧹🧹🧹 

I added some questions to clarify test coverage because the PR description is a bit short how we cover these.
Just wondering for context, were these old tests?

File: tests/aws/services/events/test_events_rules.py [hunk 4]
```diff
@@ -1,345 +0,0 @@
...
-        ]
-    )
-
-    event_detail_exists = {"key": "value", "payload": "baz"}
-    event_detail_not_exists = {"no-key": "no-value", "payload": "baz"}
-    event_patter_details = ["core.update-account-command"]
-    test_event_pattern_exists = {
-        "source": event_patter_details,
-        "detail-type": event_patter_details,
-        "detail": {"key": [{"exists": True}]},
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:128]: https://github.com/localstack/localstack/pull/11189

File: tests/aws/services/events/test_events_rules.py [hunk 5]
```diff
@@ -1,345 +0,0 @@
...
-        ]
-    )
-
-    event_detail_match = {"command": "display-message", "payload": "baz"}
-    event_detail_null = {"command": None, "payload": "baz"}
-    event_detail_no_match = {"command": "no-message", "payload": "baz"}
-    test_event_pattern_anything_but = {
-        "source": ["core.update-account-command"],
-        "detail-type": ["core.update-account-command"],
-        "detail": {"command": [{"anything-but": ["no-message"]}]},
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:72]: https://github.com/localstack/localstack/pull/11189

File: tests/aws/services/events/test_events_rules.py [hunk 6]
```diff
@@ -1,345 +0,0 @@
...
-        ]
-    )
-
-    event_detail_exists = {"key": "value", "payload": "baz"}
-    event_detail_not_exists = {"no-key": "no-value", "payload": "baz"}
-    event_patter_details = ["core.update-account-command"]
-    test_event_pattern_not_exists = {
-        "source": event_patter_details,
-        "detail-type": event_patter_details,
-        "detail": {"key": [{"exists": False}]},
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:175]: https://github.com/localstack/localstack/pull/11189

File: tests/aws/services/events/test_events_rules.py (hunk 1 above)
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:308]: Yes the cloudwatch scenario is still not covered

File: tests/aws/services/events/test_events_rules.py [hunk 7]
```diff
@@ -1,345 +0,0 @@
...
-
-    messages_not_exists = put_events_with_filter_to_sqs(
-        pattern=test_event_pattern_not_exists,
-        entries_asserts=entries_asserts_exists_false,
-    )
-    snapshot.match("rule-exists-false", messages_not_exists)
-
-
-@markers.aws.unknown
-@pytest.mark.skipif(is_v2_provider(), reason="V2 provider does not support this feature yet")
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:204]: https://github.com/localstack/localstack/pull/11189

File: tests/aws/services/events/test_events_rules.py [hunk 8]
```diff
@@ -1,345 +0,0 @@
...
-def test_put_event_with_content_base_rule_in_pattern(aws_client, clean_up):
-    queue_name = f"queue-{short_uid()}"
-    rule_name = f"rule-{short_uid()}"
-    target_id = f"target-{short_uid()}"
-    event_bus_name = f"event-bus-{short_uid()}"
-
-    queue_url = aws_client.sqs.create_queue(QueueName=queue_name)["QueueUrl"]
-    queue_arn = arns.sqs_queue_arn(queue_name, TEST_AWS_ACCOUNT_ID, TEST_AWS_REGION_NAME)
-
-    # EventBridge apparently converts some fields, for example: Source=>source, DetailType=>detail-type
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events_rules.py:214]: https://github.com/localstack/localstack/pull/11189

**Review by joe4dev — APPROVED** [review]
Thanks for the clarification. I got confused because these tests were added within another PR and some renamed at the same time. For example:

* test_put_event_with_content_base_rule_in_pattern in test_events_rules (deleted here) => test_put_event_with_content_base_rule_in_pattern in test_events_patterns (added in https://github.com/localstack/localstack/pull/11189)
* test_event_pattern_anything_but in test_events_rules (deleted here) => test_put_events_with_rule_pattern_anything_but in test_events_patterns (added in https://github.com/localstack/localstack/pull/11189)

Let's try to keep each test refactoring fully contained within the same PR because it's very hard to track copy (as duplicate) within one PR followed by a delete of the same test within another PR. We have more details on our "Git(Hub) Best Practices" page in Notion under "… scope PRs to preserve the git file history".

**Comment by joe4dev** [comment]
Re-triggered the failing Build in CircleCI (looks like a temporary build issue):
Screenshot 2024-07-23 at 11 54 29

**Pull request merged**